### ISSUE TITLE
MultipleConsumers command addtional argument

### ISSUE THREAD

**Issue by tiborb** [php-amqplib/RabbitMqBundle#411]
MultipleConsumers suppiot a queue provider service, but it is not possible to provider the underlying service with and argument to change the queue settings. I suggest adding a additional argument just like in the DynamicConsumers command.

The final command would look like:

```
php app/console  rabbitmq:multiple-consumer example_consumer_name special_arg
```

Any thoughts?

Cheers,
Tibor

**Comment by skafandri** [comment]
What about a more generic approach to make this feature available to all consumers? Did you thought about a solution already?

**Comment by tiborb** [comment]
@skafandri not really.
Here is a list of consumers that could profit from this feature:
```
rabbitmq:anon-consumer
rabbitmq:consumer
rabbitmq:dynamic-consumer
rabbitmq:multiple-consumer
```

**Comment by tiborb** [comment]
Hi @skafandri, after some research, a general solution would involve a dramatic refactoring of the consumers, resulting in several BCs.

For the major change, I would suggest always loading the queue and consumer settings using a service which will provide the default settings if no special argument was be provided. More or less a v2.0.0

For the v1.x series i would just implement this minor change in the multiple consumers as specified before.

**Comment by skafandri** [comment]
We can leave a better solution to v2 then. The proposed implementation should be good enough for now. If we change `context` to an `InputOption` of type array it wouldn't be more flexible?

**Comment by tiborb** [comment]
Salut, option would be possible as well, but was thinking about an argument analog to the dynamic-consumers. But optional. What to you think?

**Comment by skafandri** [comment]
I just saw the dynamic consumer implementation. Yes, consistency is the most important :)

**Comment by tiborb** [comment]
@skafandri ok, than i will add it a test tomorrow and pull it. How about we do a minor than, say v1.12.0?

**Comment by skafandri** [comment]
I think so, there are already couple of small features and fixes on master. I would like to start using them.

**Comment by tiborb** [comment]
merged in 0bb11edec0abdf6dc28eac0144c0a673dddba28f
